pkg/registry/resourcequota/strategy.go

```diff
@@ -106,9 +106,8 @@ func MatchResourceQuota(label labels.Selector, field fields.Selector) generic.Ma
 }
 
 // ResourceQuotaToSelectableFields returns a label set that represents the object
-// TODO: fields are not labels, and the validation rules for them do not apply.
 func ResourceQuotaToSelectableFields(resourcequota *api.ResourceQuota) labels.Set {
 	return labels.Set{
-		"name": resourcequota.Name,
+		"metadata.name": resourcequota.Name,
```

Comment:
Actually I dont see any field label conversion funcs defined for ResourceQuotas: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/conversion.go#L41.
Indeed ```api/v1/watch/namespaces/default/resourcequotas?fieldSelector=name=foo``` returns:
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "No field label conversion function found for version v1 and kind ResourceQuota",
  "reason": "BadRequest",
  "code": 400
}
```

So this change will not break any external client using the REST API. This will still, however, break internal code that is directly calling storage.Watch(labels.Everything, fields.AsSelector(fields.Set{"name":"foo"})).
I guess, we should be fine as long as all our tests pass.